lib/netif.ml

```diff
@@ -67,6 +67,7 @@ let err_partial_write len' page =
 
 let connect devname =
   try
+    Random.self_init ();
     let fd, devname = Tuntap.opentap ~pi:false ~devname () in
     let dev = Lwt_unix.of_unix_file_descr ~blocking:true fd in
     let mac = Macaddr.make_local (fun _ -> Random.int 256) in
```

Comment:
FYI some MAC related stuff here:
https://github.com/xapi-project/xenopsd/blob/900f9ec3251fd9850a1326a3617a73639a10c01c/lib/mac.ml#L32

On Mon, May 16, 2016 at 4:31 PM, Mindy Preston <notifications@github.com>
wrote:

> In lib/netif.ml
> <https://github.com/mirage/mirage-net-unix/pull/24#discussion_r63372705>:
>
> > @@ -67,6 +67,7 @@ let err_partial_write len' page =
> >
> >  let connect devname =
> >    try
> > +    Random.self_init ();
> >      let fd, devname = Tuntap.opentap ~pi:false ~devname () in
> >      let dev = Lwt_unix.of_unix_file_descr ~blocking:true fd in
> >      let mac = Macaddr.make_local (fun _ -> Random.int 256) in
>
> On second thought, a completely random MAC is a bad idea. MAC addresses
> are structured and we should make sure that the globally-unique/link-local
> and unicast/multicast bits of the OUI are set as expected.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mirage/mirage-net-unix/pull/24/files/b3aed3776ca65fbe6da916755292b6163d54c181#r63372705>
>



-- 
Dave Scott